ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -352,3 +373,67 @@ instance ( ShelleyBasedEra era
         Comp . WrapValidatedGenTx
       . mkShelleyValidatedTx . SL.coerceValidated
     <$> SL.translateValidated @era @WrapTx ctxt (SL.coerceValidated vtx)
+
+{-------------------------------------------------------------------------------
+  Canonical TxIn
+-------------------------------------------------------------------------------}
+
+instance ShelleyBasedEra era
+      => HasCanonicalTxIn '[ShelleyBlock proto era] where
+  newtype instance CanonicalTxIn '[ShelleyBlock proto era] = ShelleyBlockHFCTxIn {
+      getShelleyBlockHFCTxIn :: SL.TxIn (EraCrypto era)
+    }
+    deriving stock (Show, Eq, Ord)
+    deriving newtype NoThunks
+
+  injectCanonicalTxIn IZ txIn     = ShelleyBlockHFCTxIn txIn
+  injectCanonicalTxIn (IS idx') _ = case idx' of {}
+
+  distribCanonicalTxIn IZ txIn     = getShelleyBlockHFCTxIn txIn
+  distribCanonicalTxIn (IS idx') _ = case idx' of {}
+
+  encodeCanonicalTxIn (ShelleyBlockHFCTxIn txIn) = SL.toEraCBOR @era txIn
+
+  decodeCanonicalTxIn = ShelleyBlockHFCTxIn <$> SL.fromEraCBOR @era
+
+{-------------------------------------------------------------------------------
+  HardForkTxOut
+-------------------------------------------------------------------------------}
+
+instance HasHardForkTxOut '[ShelleyBlock proto era] where
+  type instance HardForkTxOut '[ShelleyBlock proto era] = SL.TxOut era
+  injectHardForkTxOut IZ txOut    = txOut
+  injectHardForkTxOut (IS idx') _ = case idx' of {}
+  distribHardForkTxOut IZ txOut    = txOut
+  distribHardForkTxOut (IS idx') _ = case idx' of {}
+
+instance ShelleyBasedEra era => SerializeHardForkTxOut '[ShelleyBlock proto era] where
+  encodeHardForkTxOut _ = SL.toEraCBOR @era
+  decodeHardForkTxOut _ = SL.fromEraCBOR @era
```

Comment:
Perhaps this should be `fromEraShareCBOR`, same as in the normal shelley block. There are no mainnet ledgerstates of the unary shelley HF block, so it doesn't really matter I think.

Reply:
Couldn't we just defer to the `CanSerializeLedgerTables (LedgerState (ShelleyBlock proto era))` instance to use the correct serialization automatically?

We could even use `DefaultHardForkTxOut` for that I think.

Reply:
Lets lift this into an issue.

Reply:
Falls into this issue https://github.com/IntersectMBO/ouroboros-consensus/issues/1309